utils/azureCredentialsManager.ts

```diff
@@ -0,0 +1,140 @@
+import { SubscriptionClient, ResourceManagementClient, SubscriptionModels } from 'azure-arm-resource';
+import { AzureAccount } from '../typings/azure-account.api';
+import { ServiceClientCredentials } from 'ms-rest';
+import { AsyncPool } from '../utils/asyncpool';
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import * as ContainerModels from '../node_modules/azure-arm-containerregistry/lib/models';
+import { ResourceGroup, ResourceGroupListResult } from "azure-arm-resource/lib/resource/models";
+import { MAX_CONCURRENT_SUBSCRIPTON_REQUESTS } from './constants';
+
+/* Singleton for facilitating communication with Azure account services by providing extended shared
+  functionality and extension wide access to azureAccount. Tool for internal use.
+  Authors: Esteban Rey L, Jackson Stokes
+*/
+
+export class AzureCredentialsManager {
+
+    //SETUP
+    private static _instance: AzureCredentialsManager = new AzureCredentialsManager();
```

Comment:
Do no initialize here, defeats the purpose of lazy initialization.

Reply:
Oh that's my bad, I thought Id gotten that fixed.